server/src/main/java/org/elasticsearch/gateway/MetaStateService.java

```diff
@@ -112,34 +174,115 @@ MetaData loadGlobalState() throws IOException {
     }
 
     /**
-     * Writes the index state.
+     * Writes manifest file (represented by {@link MetaState}) to disk.
      *
+     * @throws WriteStateException if exception when writing state occurs. See also {@link WriteStateException#isDirty()}
+     */
+    public long writeMetaState(String reason, MetaState metaState) throws WriteStateException {
+        logger.trace("[_meta] writing state, reason [{}]", reason);
+        try {
+            long generation = MetaState.FORMAT.write(metaState, nodeEnv.nodeDataPaths());
+            logger.trace("[_meta] state written (generation: {})", generation);
+            return generation;
+        } catch (WriteStateException ex) {
+            logger.warn("[_meta]: failed to write meta state", ex);
```

Comment:
I think it'd be better to log this where the exception is ultimately caught.

Reply:
Wrapping WriteStateException and with the the message specifying what exactly has failed should address this comment.